truffle/com.oracle.truffle.tools.debug.shell/src/com/oracle/truffle/tools/debug/shell/client/REPLRemoteCommand.java

```diff
@@ -579,6 +579,7 @@ void processReply(REPLClientContext context, REPLMessage[] replies) {
             } else {
                 context.displayFailReply(replies[0].get(REPLMessage.DISPLAY_MSG));
             }
+            throw new REPLContinueException();
```

Comment:
Exactly.  It wasn't a bug. The code was interrupted by KillException before. After we prepare for the kill only, we need to continue to actually finish the kill.